### PR TITLE
fix(async-queuer): use nullish check in #tick to handle falsy queue items

### LINKED ISSUE #200: AsyncQueuer silently drops falsy queue items (0, "", false, null)
### TanStack Pacer version

@tanstack/pacer 0.20.1 (current latest)

### Framework/Library version

Node 24

### Describe the bug and the steps to reproduce it

`AsyncQueuer` silently drops falsy queue items. Values like `0`, `""`, `false`, and `null` are accepted into the queue via `addItem()`, but they are never processed by the `#tick()` loop.

### Root Cause

In `packages/pacer/src/async-queuer.ts`, the `#tick()` method uses a falsy check when peeking at the next item:
```ts
const nextItem = this.peekNextItem()
if (!nextItem) {
  break
}
```

### Your Minimal, Reproducible Example - (Sandbox Highly Recommended)

https://stackblitz.com/edit/stackblitz-starters-uqnwpqze?file=index.js

### Screenshots or Videos (Optional)

_No response_

### Do you intend to try to help solve this bug with your own PR?

Yes, I am also opening a PR that solves the problem along side this issue

### Terms & Code of Conduct

- [x] I agree to follow this project's Code of Conduct
- [x] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

### PR DESCRIPTION
**Pull request by simonmeyerrr**
## 🎯 Changes

Fixes https://github.com/TanStack/pacer/issues/200

`AsyncQueuer#tick()` used `if (!nextItem)` to check whether the queue was empty, which caused falsy items (`0`, `""`, `false`) to be silently skipped. Changed to `if (nextItem === undefined)` to match the sentinel already used by `getNextItem()` and `execute()`.

Added tests verifying that items with values `0`, `""`, and `false` are processed correctly.

The issue is actually deeper than this, but this PR aim to make the minimal fix for this cases. A better fix would be to refactor some methods signature from `TValue | undefined` to something like `{ hasItem: false } | { hasItem: true, item: TValue }` in order to also allow `undefined` as a value (as nothing in `addItem` or `TValue` prevent having an undefined item), but it would require a major version change.

Note: I didn't add a test for `null`, because this value makes other part of the code crash (priority computing) which should be fixed in other issues/PRs.

## ✅ Checklist

- [x] I have followed the steps in the [Contributing guide](https://github.com/TanStack/pacer/blob/main/CONTRIBUTING.md).
- [x] I have tested this code locally with `pnpm run test:pr`.

## 🚀 Release Impact

- [x] This change affects published code, and I have generated a [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).
- [ ] This change is docs/CI/dev-only (no release).

## Bug Fixes
* Fixed an issue where queued items with falsy values (`0`, empty string, `false`) were incorrectly excluded from processing, ensuring they are now properly handled and processed as intended.

## Tests
* Added test coverage for falsy item handling to verify correct processing of items with falsy values.

### CODE REVIEW

**Comment by simonmeyerrr** [comment]
> 🧹 Nitpick comments (1)

Skipping `null` here, because it still blows up in priority handling on add (`(item as any).priority`), which is a separate issue. `0`/`""`/`false` already lock the `#tick` sentinel fix. `NaN` isn’t needed for this patch.

File: packages/pacer/src/async-queuer.ts [hunk 1]
```diff
@@ -437,7 +437,7 @@ export class AsyncQueuer<TValue> {
       this.store.state.items.length > 0
     ) {
       const nextItem = this.peekNextItem()
-      if (!nextItem) {
+      if (nextItem === undefined) {
         break
       }
       activeItems.push(nextItem)
```
[the remark this answers is not in the document]
Comment (simonmeyerrr) [reply packages/pacer/src/async-queuer.ts:443]: Agreed this look like a real pre-existing mismatch (`peekNextItem()` defaults to `'front'` while `execute()` uses `getItemsFrom`), but it’s unrelated to the falsy-sentinel change in this PR.

File: packages/pacer/tests/async-queuer.test.ts [hunk 2]
```diff
@@ -1095,4 +1095,70 @@ describe('AsyncQueuer', () => {
...
+      asyncQueuer.addItem(true)
+      asyncQueuer.addItem(false)
+      asyncQueuer.start()
+
+      await vi.advanceTimersByTimeAsync(100)
+
+      expect(processed).toEqual([false, true, false])
+      expect(asyncQueuer.store.state.successCount).toBe(3)
+    })
+
```
[the remark this answers is not in the document]
Comment (simonmeyerrr) [reply packages/pacer/tests/async-queuer.test.ts:1162]: Same as above: a `getItemsFrom: 'back'` case would mainly catch the peek/dequeue alignment bug above, not this regression. Leaving it out of this PR.

**Comment by KevinVandy** [comment]
gave you co-auther credit in a larger PR that fixed this and even more issues at all once
